### PR TITLE
Add client photo permission

### PR DESCRIPTION
**Pull request by martha**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

PT ticket: 
https://www.pivotaltracker.com/story/show/187308072

This PR adds a permission for viewing client photos.

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/spec/requests/hmis/lookup_client_spec.rb [hunk 1]
```diff
@@ -146,6 +155,9 @@
       'addresses' => [{ 'line1' => '999 Test Ave' }],
       'phoneNumbers' => [{ 'value' => '5554567891', 'use' => 'home', 'system' => 'phone' }],
       'emailAddresses' => [{ 'value' => 'email@e.mail', 'use' => 'home', 'system' => 'email' }],
+      'image' => {
+        # TODO @MARTHA ???
```
Comment (martha) [inline drivers/hmis/spec/requests/hmis/lookup_client_spec.rb]: TO DO: Fix this test. I spent some time stuck on this and wasn't able to figure out why tags were not getting saved. 

Paused at a breakpoint in the test,
```
0> GrdaWarehouse::ClientFile.count 
=> 1

0> GrdaWarehouse::ClientFile.first.tag_list 
=> 
# why empty?

0> photo.tag_list
=> Client Headshot
# not empty...

0> GrdaWarehouse::ClientFile.first.id 
=> 307243

0> photo.id 
=> 307243
# IDs match

0> photo.tags.count 
=> 0

0> GrdaWarehouse::ClientFile.first.tags.count
=> 0
# both zero :(
```

**Comment by gigxz** [comment]
oops I'm sorry @martha I didn't realize this was opened against the other branch, and now I already merged it in. can you address the conflicts and update? thank you!!

File: drivers/hmis/spec/requests/hmis/lookup_client_spec.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/spec/requests/hmis/lookup_client_spec.rb]: not sure where your breakpoint was @martha but I put one here, and saw that the `GrdaWarehouse::ClientFile.client_photos` scope is empty https://github.com/greenriver/hmis-warehouse/blob/634e0e6685e41c7d9f328e1860c8e8779d185fdb/drivers/hmis/app/graphql/types/hmis_schema/client.rb#L243

Comment (gigxz) [reply drivers/hmis/spec/requests/hmis/lookup_client_spec.rb]: Oh, I think you just need to associate the file with the client you're resolving:
`let!(:photo) { create :client_file }`
should be:
`let!(:photo) { create :client_file, client: my_client }`

Comment (martha) [reply drivers/hmis/spec/requests/hmis/lookup_client_spec.rb]: Thanks for looking at it, @gigxz ! I added a test above on line 142 that illustrates the issue a bit more clearly. Line 143 passes but line 144 fails - the scope is still not returning anything even when (I think) the file is tagged correctly.

Comment (gigxz) [reply drivers/hmis/spec/requests/hmis/lookup_client_spec.rb]: ah ok. I was tracing through the code and saw this comment on another function :\
https://github.com/greenriver/hmis-warehouse/blob/af5baa14aa2b0b5bca9009688ac6ead9e91594d7/app/models/grda_warehouse/client_file.rb#L129

The tagged_with fn comes from this gem https://github.com/mbleigh/acts-as-taggable-on. It looks like these comments are very old so I wonder if this is still the case. @eanders do you have any background here?


I do think its important to be able to test photo resolution in the api, so maybe this function needs to be replaced with whatever the test-friendly version is. lets wait to hear from Elliot though.

https://github.com/greenriver/hmis-warehouse/blob/af5baa14aa2b0b5bca9009688ac6ead9e91594d7/app/models/concerns/client_file_base.rb#L26-L28

Comment (martha) [reply drivers/hmis/spec/requests/hmis/lookup_client_spec.rb]: Ah, thank you @gigxz , that's good to know. I played around with it and could, indeed, get it working in the rspec test by avoiding `tagged_with`, which looks like this: https://github.com/greenriver/hmis-warehouse/pull/4222/files#diff-cfe82ea280b947f0dbbffd0f15c9e9b3fb05d9d36794e210390b315413a1ceddR27-R35
I'd be happy to implement some common solution though if we are now making this accommodation in multiple places, but yes would like to wait for @eanders's input on how to proceed.

File: drivers/hmis/spec/requests/hmis/lookup_client_spec.rb [hunk 2]
```diff
@@ -147,6 +154,7 @@
       'phoneNumbers' => [{ 'value' => '5554567891', 'use' => 'home', 'system' => 'phone' }],
       'emailAddresses' => [{ 'value' => 'email@e.mail', 'use' => 'home', 'system' => 'email' }],
     )
+    expect(result.dig('data', 'client', 'image')).not_to be_nil
   end
```
Comment (gigxz) [inline drivers/hmis/spec/requests/hmis/lookup_client_spec.rb:159]: Can you add a test where it removes `can_view_client_photo` and ensures the photo is not resolved?

File: spec/models/grda_warehouse/client_file_spec.rb [hunk 3]
```diff
@@ -4,9 +4,11 @@
   describe 'Creating a client file' do
     let!(:consent_tag) { create :available_file_tag, consent_form: true, name: 'Consent Form', full_release: true }
     let!(:other_tag) { create :available_file_tag, consent_form: false, name: 'Other Tag' }
+    let!(:photo_tag) { create :available_file_tag, consent_form: false, name: 'Client Headshot' }
     let!(:file) { create :client_file, effective_date: 5.days.ago }
     let!(:second_file) { create :client_file, effective_date: 3.days.ago, client: file.client }
     let!(:third_file) { create :client_file, effective_date: 1.days.ago, client: file.client }
+    let!(:photo_file) { create :client_file, effective_date: 1.days.ago, client: file.client }
```
Comment (gigxz) [inline spec/models/grda_warehouse/client_file_spec.rb]: Whats the purpose of the new lines in this test file?

File: spec/factories/grda_warehouse/client_files.rb [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
+
     association :client, factory: :grda_warehouse_hud_client
     association :user, factory: :user
     client_file { Rack::Test::UploadedFile.new('spec/fixtures/files/images/test_photo.jpg', 'image/jpg') }
     name { 'Test File' }
     visible_in_window { true }
+
+    before(:create) do |file, evaluator|
+      file.tag_list = evaluator.tags.map(&:name)
+    end
```
Comment (gigxz) [inline spec/factories/grda_warehouse/client_files.rb:15]: Nice. I am a big fan of the transient attributes. Much cleaner than the alternative (eg:
https://github.com/greenriver/hmis-warehouse/blob/e8f882c80e860b837b3acb99e7194d6ebdbcf53a/spec/models/grda_warehouse/client_file_legacy_spec.rb#L170-L177

**Review by gigxz — COMMENTED** [review]
Looks great! Just one request for an additional test case.
I'll request review from Elliot too since this touches some Warehouse code and factories.

**Review by eanders — APPROVED** [review]

File: spec/models/grda_warehouse/client_file_spec.rb (hunk 3 above)
Comment (martha) [reply spec/models/grda_warehouse/client_file_spec.rb]: Ah sorry this is just an artifact from me hacking around trying to get the tagging to work correctly in tests. I'll remove them.

File: drivers/hmis/spec/requests/hmis/lookup_client_spec.rb (hunk 2 above)
Comment (martha) [reply drivers/hmis/spec/requests/hmis/lookup_client_spec.rb:159]: Ah, yeah of course! Sorry I forgot this.

**Review by gigxz — APPROVED** [review]

**Pull request merged**